packages/repository/src/relations/belongs-to/belongs-to.decorator.ts

```diff
@@ -53,3 +58,22 @@ export function belongsTo<T extends Entity>(
     relation(meta)(decoratedTarget, decoratedKey);
   };
 }
+
+/**
+ * Sugar syntax for belongsTo decorator for hasOne relation. Calls belongsTo
+ * with property definition defaults for non-generated id field
+ * @param targetResolver A resolver function that returns the target model for
+ * a belongsTo relation
+ * @param definition Optional metadata for setting up a belongsTo relation
+ * @returns {(target: Object, key:string)}
+ */
+export function belongsToUniquely<T extends Entity>(
+  targetResolver: EntityResolver<T>,
+  definition?: Partial<BelongsToDefinition>,
+) {
+  const propertyMetaDefaults: Partial<PropertyDefinition> = {
+    id: true,
+    generated: false,
+  };
```

Comment:
Is the intention to make the FK unique? If so, I think it's wrong to set `id` to `true` as it will add the FK to the PK. For example, `(id, customerId)` becomes the PK.

I think we need to mark it as `unique`. IIRC, we can mark it uniquely indexed.

Reply:
> Is the intention to make the FK unique?

Yes.

> If so, I think it's wrong to set id to true as it will add the FK to the PK. For example, (id, customerId) becomes the PK.

The way we have envisioned the current hasOne implementation, there is only one PK property, and that's `customerId`, there is no property called `id`. I.e. the primary key is a foreign key too.

> I think we need to mark it as unique. IIRC, we can mark it uniquely indexed.

I am not sure if all databases support `unique` constraint. I am also pretty sure our auto-migration implementation does not correctly define `unique` constraint either.

As a result, by using `unique` instead of `id`, the default implementation of `hasOne` in LB4 would not work correctly out of the box.

On the other hand, all databases are enforcing the uniqueness constraint on primary keys (id property), otherwise primary keys could not be used as unique identifiers.

Reply:
I think it's too restrictive to promote the FK for `hasOne` to be part of PK. At least we should check if there is an existing PK.

BTW, marking a property `id: true` does not guarantee it will be enforced by the database. I think we should generate an index with `{unique: true}` option per https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html#indexes.

Reply:
Not only we are promoting FK to be the only PK, but we also require the target od HasOne to have the same PK as the model it relates to. E.g. when a Customer has one PublicProfile, then both the customer instance and the profile instance share the same primary key value.

This is the only way I am aware of that offers strong guarantees about consistency and avoid possible race conditions in all databases we support in LoopBack.

I am more than happy to consider different approaches, as long as they work similarly well across the databases we have connectors for.

Should we open a new issue to continue this discussion?